soroban-sdk/src/tests/token_client.rs

```diff
@@ -48,7 +48,12 @@ fn test_mock_all_auth() {
     let env = Env::default();
 
     let admin = Address::generate(&env);
-    let token_contract_id = env.register_stellar_asset_contract(admin);
+    let sac = env.register_stellar_asset_contract_v2(admin);
+    let token_contract_id = sac.address.clone();
+
+    assert_eq!(sac.get_issuer_flags(), 0);
+    sac.overwrite_issuer_flags(1);
+    assert_eq!(sac.get_issuer_flags(), 1);
```

Comment:
With the changes I suggest above the example should look something like:
```suggestion
    let sac = env.register_stellar_asset_contract(admin.clone());
    let issuer = sac.issuer();

    assert_eq!(issuer.flags(), 0);
    issuer.set_flags(1);
    assert_eq!(issuer.flags(), 1);
```